clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -248,9 +239,9 @@ def _get_pods_by_ip_or_pending(
             if self._pod_belongs_to_pool(pod):
                 if exclude_daemonset_pods and self._pod_belongs_to_daemonset(pod):
                     excluded_pods_by_ip[pod.status.host_ip].append(pod)
-                elif pod.status.phase == "Running":
+                elif pod.status.phase == "Running" or self._is_pod_scheduled(pod):
                     pods_by_ip[pod.status.host_ip].append(pod)
-                else:
+                elif self._is_pod_unschedulable(pod):
```

Comment:
do we need this extra method? if a Pod doesn't belong to a daemonset (in any phase) or if a Pod isn't running/about to be running, it can only be pending, right? or were we getting error/terminated/etc pods here and counting these as pending before?

Reply:
it can be Succeeded and Failed for jobs. that's the main reason of phase check.
And "pending" phase doesn't mean that pod is unschedulable: `The Pod has been accepted by the Kubernetes cluster, but one or more of the containers has not been set up and made ready to run. This includes time a Pod spends waiting to be scheduled as well as the time spent downloading container images over the network`

If I understood correctly we only care about unschedulable pods, Because we use "_pending_pods" only for "get_unschedulable_pods" method. So pending_pods=unschedulable_pods.
What do you think?